lib/trino-hdfs/src/main/java/io/trino/hdfs/util/NodeProvider.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.hdfs.util;
+
+import io.trino.spi.HostAddress;
+
+import java.util.Optional;
+
+public interface NodeProvider
```

Comment:
Perhaps this should be renamed `CacheDistribution` or something similar? E.g., there could be caching strategies where all data is cached on all nodes, or where subsets of the data are cached on all nodes, while other subsets of the data is cached in a round robin fashion, or never cached. This interface really tells how the cache is distributed among the nodes.

There might also need to be a method `cachedBy(String identifer, HostAddress address)` to allow a host to determine if it is supposed to cache a particular piece of data.